.github/workflows/publish-version.yml

```diff
@@ -543,86 +472,37 @@ jobs:
 
-      - name: Patch crate version
-        if: ${{ needs.prepare-vars.outputs.environment != 'stable' }}
+      - name: Patch crate name, version, and description
         run: |
           set -x
 
           # Derive crate version
-          currentVersion=${{ needs.prepare-vars.outputs.current-version }}
-          major=$(echo $currentVersion | tr "." "\n" | sed -n 1p)
-          minor=$(echo $currentVersion | tr "." "\n" | sed -n 2p)
-          version=${major}.${minor}.${{ needs.prepare-vars.outputs.patch }}
-
-          # Update the version to a nightly one
-          sed -i "s#^version = \".*\"#version = \"${version}\"#" Cargo.toml
-          sed -i "s#^version = \".*\"#version = \"${version}\"#" crates/sdk/Cargo.toml
-          sed -i "s#^version = \".*\"#version = \"${version}\"#" crates/core/Cargo.toml
-          sed -i "s#surrealdb = { version = \"=${{ needs.prepare-vars.outputs.version }}\"#surrealdb = { version = \"=${version}\"#" Cargo.toml
-          sed -i "s#surrealdb-core = { version = \"=${{ needs.prepare-vars.outputs.version }}\"#surrealdb-core = { version = \"=${version}\"#" Cargo.toml
+          ENVIRONMENT=${{ needs.prepare-vars.outputs.environment }} \
+            CURRENT_VERSION=${{ needs.prepare-vars.outputs.current-version }} \
+            VERSION=${{ needs.prepare-vars.outputs.version }} \
+            PATCH=${{ needs.prepare-vars.outputs.patch }} \
+            ./.github/scripts/patch_crate.sh all
 
           # Temporarily commit patch
-          git commit -am "Patch crate version"
+          git commit -am "Patch crate name, version, and description"
 
-      - name: Patch crate name and description
-        if: ${{ needs.prepare-vars.outputs.environment != 'stable' }}
-        run: |
-          set -x
-
-          environment=${{ needs.prepare-vars.outputs.environment }}
-
-          # Patch crate name
-          sed -i "0,/surrealdb/s//surrealdb-${environment}/" crates/sdk/Cargo.toml
-          sed -i "0,/surrealdb-core/s//surrealdb-core-${environment}/" crates/core/Cargo.toml
-
-          # Patch dependency package
-          sed -i "s/package = \"surrealdb\"/package = \"surrealdb-${environment}\"/" Cargo.toml
-          sed -i "s/package = \"surrealdb-core\"/package = \"surrealdb-core-${environment}\"/" Cargo.toml
-
-          # Patch the description
-          if [[ $environment == 'alpha' ]]; then
-            start="An"
-          else
-            start="A"
-          fi
-          sed -i "s#^description = \".*\"#description = \"${start} ${environment} release of the surrealdb crate\"#" crates/sdk/Cargo.toml
-          sed -i "s#^description = \".*\"#description = \"${start} ${environment} release of the surrealdb-core crate\"#" crates/core/Cargo.toml
-
-          # Temporarily commit patch
-          git commit -am "Patch crate names"
-
-      - run: cat Cargo.toml
-      - run: cat crates/sdk/Cargo.toml
-      - run: cat crates/core/Cargo.toml
+      - run: cat Cargo.toml ./crates/*/Cargo.toml
```

Comment:
I think I prefer the separate steps. They make it quicker to jump to the Cargo.toml file you want to see and only expand that.